sysl2/sysl/syslutil/chroot_fs.go

```diff
@@ -70,3 +78,10 @@ func (fs *ChrootFs) Chmod(name string, mode os.FileMode) error {
 func (fs *ChrootFs) Chtimes(name string, atime time.Time, mtime time.Time) error {
 	return fs.fs.Chtimes(fs.join(name), atime, mtime)
 }
+
+func (fs *ChrootFs) openAllowed(name string) (err error) {
+	if strings.Contains(name, "..") {
+		return errors.New("permission denied, file outside root")
+	}
+	return
+}
```

Comment:
`abc/../..def` is valid.

Reply:
If filepath.Clean() is used the path will never have .. in it anywah

Reply:
Only inner refs such as `abc/..` will be collapsed. Prefix `../` will remain, so the solution is to apply `filepath.Clean` and then error out if `name == ".." || strings.HasPrefix(name, "../)`.

Reply:
resolved in the test